### PR TITLE
Tensorflow addons

### PR DESCRIPTION
**Pull request by bhack**
Add support to Tensorflow addons:

See https://github.com/tensorflow/addons/pull/1309 and https://github.com/tensorflow/addons/pull/1888

### CODE REVIEW

**Comment by bhack** [comment]
As you have seen in the mentioned ticket we would like to have CPU and GPU separated configs but I think is not possible currently if you check https://github.com/microsoft/vscode-remote-release/issues/2143#issuecomment-600087021

File: repository-containers/github.com/tensorflow/addons/.devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# property in devcontainer.json to use it. On Linux, the container user's GID/UIDs
+# will be updated to match your local UID/GID (when using the dockerFile property).
+# See https://aka.ms/vscode-remote/containers/non-root-user for details.
+ARG USERNAME=vscode
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+
+# Options for common package install script - SHA updated on release
+ARG INSTALL_ZSH="false"
+ARG UPGRADE_PACKAGES="true"
```
Comment (bhack) [inline repository-containers/github.com/tensorflow/addons/.devcontainer/Dockerfile:18]: @seanpmorgan Do we want to update the package by default or not?

Comment (seanpmorgan) [reply repository-containers/github.com/tensorflow/addons/.devcontainer/Dockerfile:18]: I'd prefer they not be updated in this Dockerfile but rather done in tfaddons/dev_container if needed

**Comment by bhack** [comment]
We are waiting to check test and test discover to see if we could enable `"python.testing.pytestEnabled": true,`

**Comment by bhack** [comment]
@Chuxel is there a way to differentiate between CPU and GPU `.devcontainer`? Or are we still blocked by https://github.com/microsoft/vscode-remote-release/issues/2143?

**Comment by Chuxel** [comment]
@bhack I would use a build arg if you need to modify the contents of the Dockerfile and add a comment into devcontainer.json about switching between the two.

**Comment by bhack** [comment]
@Chuxel need we to comment and uncomment also https://github.com/microsoft/vscode-remote-release/issues/345#issuecomment-493271078 right?

**Comment by Chuxel** [comment]
@bhack If there's arguments you need to pass in, yes.  The .NET Core definition is an example of this being done - in this case for SSL support: https://github.com/microsoft/vscode-dev-containers/blob/master/containers/dotnetcore-3.1/.devcontainer/devcontainer.json

File: repository-containers/github.com/tensorflow/addons/.devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	"dockerFile": "Dockerfile",
+	
+    	// Uncomment to enable Nvidia support
+	// "runArgs": ["--runtime=nvidia"],
+
+	// Set *default* container specific settings.json values on container create.
+	"settings": { 
+		"terminal.integrated.shell.linux": null,
+		"python.formatting.provider": "black",
+		"python.testing.pytestEnabled": true,
```
Comment (bhack) [inline repository-containers/github.com/tensorflow/addons/.devcontainer/devcontainer.json:13]: @seanpmorgan Let me know if we could enable pytest by default or the discovery it is not working.

**Comment by bhack** [comment]
@Chuxel I have another question how we could automate this manual commenting/uncommenting activity when we let the user to use this for Github codespaces?

**Comment by Chuxel** [comment]
VS Codespaces (and by extension Codespaces in GitHub) will be implementing the ability to iterate on a devcontainer.json and Dockerfile like the Remote - Containers extension over time. It's just not quite there yet.

**Comment by bhack** [comment]
@Chuxel do you mean that they have in the roadmap multiple alternative devcontainer.json and Dockerfile support for the same repo?

**Comment by Chuxel** [comment]
@bhack You'll be able to uncomment and just rebuild like you can with Remote - Containers. Step one is getting to feature parity.

**Comment by bhack** [comment]
@Chuxel Thanks. We could still host `.devcontainer` here or it will be mandatory to be hosted in our repo root for Codespaces?

**Comment by Chuxel** [comment]
Yeah, currently Codespaces does not pull from this repo, but this is part of the work we'll collaborate with that team on as iterating on a container starts to be available.  We'd want the same list of definitions from this repo there as well, so that would be the natural time to include it.

File: repository-containers/github.com/tensorflow/addons/.devcontainer/devcontainer.json (hunk 2 above)
Comment (bhack) [reply repository-containers/github.com/tensorflow/addons/.devcontainer/devcontainer.json]: The integration is commented by default cause vscode-python upstream bugs with pytest:
https://github.com/microsoft/vscode-python/issues/8678
https://github.com/microsoft/vscode-python/issues/9553

**Comment by bhack** [comment]
@seanpmorgan I've added the build-arg `IMAGE_TYPE` to switch between CPU and GPU images but we need to push a CPU image on Dockerhub.

**Comment by bhack** [comment]
@Chuxel as you see we need to comment and uncomment two sections to switch from CPU to GPU. IT is not the best UX solution but I don't find something better with the current features.

File: repository-containers/github.com/tensorflow/addons/.devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	// "build": { 
+	//		"args": { 
+	//			"IMAGE_TYPE": "latest"
+	//		} 
+	//},
+
+	"dockerFile": "Dockerfile",
+	
+	"remoteEnv": {
+		TF_CPP_MIN_LOG_LEVEL": "1"
```
Comment (Chuxel) [inline repository-containers/github.com/tensorflow/addons/.devcontainer/devcontainer.json:14]: Missing quote - `TF_CPP_MIN_LOG_LEVEL"` should be `"TF_CPP_MIN_LOG_LEVEL"`

File: repository-containers/github.com/tensorflow/addons/.devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        	"ms-vscode.cpptools"
+    	],
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	// "postCreateCommand": "uname -a",
+
+	// Uncomment to use Docker from inside the container. See https://aka.ms/vscode-remote/samples/docker-in-docker.
```
Comment (Chuxel) [inline repository-containers/github.com/tensorflow/addons/.devcontainer/devcontainer.json:45]: Is this needed in this case?  If not, you can remove these two lines.

**Comment by bhack** [comment]
It is ok for me. @seanpmorgan when you can give a pass with the GPU options we could merge this.

**Comment by Chuxel** [comment]
@seanpmorgan @bhack Let me know when you think this is to the point it should be merged.

**Comment by bhack** [comment]
> Yeah, currently Codespaces does not pull from this repo, but this is part of the work we'll collaborate with that team on as iterating on a container starts to be available. We'd want the same list of definitions from this repo there as well, so that would be the natural time to include it.

@Chuxel any news on this specific point? Where can I track progress on this? Any ticket?

**Comment by bhack** [comment]
@Chuxel gently ping

**Pull request merged**